### PR TITLE
Task/732/update subcomponent

### PR DESCRIPTION
**Pull request by hirunatan**
> Please provide enough information so that others can review your pull request:

> Explain the **details** for making this change. What existing problem does the pull request solve?

> Screenshots/GIFs

### CODE REVIEW

File: common/app/common/pages_helpers.cljc [hunk 1]
```diff
@@ -58,7 +58,7 @@
 (defn get-object-with-children
   "Retrieve a list with an object and all of its children"
   [id objects]
-  (map #(get objects %) (concat [id] (get-children id objects))))
+  (map #(get objects %) (d/concat [id] (get-children id objects))))
```
Comment (niwinz) [inline common/app/common/pages_helpers.cljc]: **minor:**  for a single element concat on start when the result type is irrelevant (this case, bacause you then using `map`, that returns a seq), the best approach is use `cons` (example: `(cons id (get-children id objects))`)

File: frontend/src/app/main/data/workspace/libraries.cljs [hunk 2]
```diff
@@ -351,26 +364,35 @@
...
-            objects        (dwc/lookup-page-objects state page-id)
-            root-id        (cph/get-root-component id objects)
-            root-shape     (get objects id)
-            file-id        (get root-shape :component-file)
-
-            components
-            (if (nil? file-id)
-              (get-in state [:workspace-data :components])
-              (get-in state [:workspace-libraries file-id :data :components]))
+      (js/console.info "##### RESET-COMPONENT of shape" (str id))
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace/libraries.cljs]: :fire:

File: frontend/src/app/main/data/workspace/libraries.cljs [hunk 3]
```diff
@@ -351,26 +364,35 @@
...
+            [all-shapes component root-component]
+            (dwlh/resolve-shapes-and-components shape
+                                                objects
+                                                state
+                                                true)
+
+            _ (js/console.info "shape" (:name shape) "<- component" (:name component))
+            _ (js/console.debug "all-shapes" (clj->js all-shapes))
+            _ (js/console.debug "component" (clj->js component))
+            _ (js/console.debug "root-component" (clj->js root-component))
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace/libraries.cljs]: :fire:

File: frontend/src/app/main/data/workspace/libraries.cljs [hunk 4]
```diff
@@ -379,60 +401,31 @@
...
-                         :name (:name component-obj)
-                         :shapes (vals component-objs)}]
-                       (map (fn [shape]
-                              {:type :mod-obj
-                               :page-id page-id
-                               :id (:id shape)
-                               :operations [{:type :set-touched
-                                             :touched (:touched shape)}]})
-                            original-shapes))]
+      (js/console.info "##### UPDATE-COMPONENT of shape" (str id))
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace/libraries.cljs]: :fire:

File: frontend/src/app/main/store.cljs [hunk 5]
```diff
@@ -89,7 +89,9 @@
      (letfn [(show-shape [shape-id level objects]
                (let [shape (get objects shape-id)]
                  (println (str/pad (str (str/repeat "  " level)
-                                        (:name shape))
+                                        (:name shape)
+                                        (when (seq (filter #(not= :position-group %)
```
Comment (niwinz) [inline frontend/src/app/main/store.cljs]: the seq + filter can be replaced with `some`

File: common/app/common/pages_helpers.cljc (hunk 1 above)
Comment (hirunatan) [reply common/app/common/pages_helpers.cljc]: fixed

File: frontend/src/app/main/store.cljs (hunk 5 above)
Comment (hirunatan) [reply frontend/src/app/main/store.cljs]: I've just realized that i removed :position-group. So it can be simplified.

**Pull request merged**